include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -50,28 +66,44 @@ namespace sparrow
         /// Neither the array nor the schema are released when the arrow_proxy is destroyed.
         explicit arrow_proxy(ArrowArray* array, ArrowSchema* schema);
 
-        ~arrow_proxy();
-
-        // TODO: implement them with the modifiers
-        arrow_proxy(const arrow_proxy&) = default;
-        arrow_proxy& operator=(const arrow_proxy&) = default;
+        // Copy constructors
+        arrow_proxy(const arrow_proxy&);
+        arrow_proxy& operator=(const arrow_proxy&);
 
+        // Move constructors
         arrow_proxy(arrow_proxy&&);
         arrow_proxy& operator=(arrow_proxy&&);
 
-        [[nodiscard]] std::string_view format() const;
-        [[nodiscard]] std::string_view name() const;
-        [[nodiscard]] std::string_view metadata() const;
+        ~arrow_proxy();
+
+        [[nodiscard]] const std::string_view format() const;
+        void set_format(const std::string_view format);
+        [[nodiscard]] enum data_type data_type() const;
+        void set_data_type(enum data_type data_type);
+        [[nodiscard]] std::optional<const std::string_view> name() const;
+        void set_name(std::optional<const std::string_view> name);
+        [[nodiscard]] std::optional<const std::string_view> metadata() const;
+        void set_metadata(std::optional<const std::string_view> metadata);
         [[nodiscard]] std::vector<ArrowFlag> flags() const;
-        [[nodiscard]] int64_t length() const;
+        void set_flags(const std::vector<ArrowFlag>& flags);
+        [[nodiscard]] size_t length() const;
+        void set_length(size_t length);
         [[nodiscard]] int64_t null_count() const;
-        [[nodiscard]] int64_t offset() const;
-        [[nodiscard]] int64_t n_buffers() const;
-        [[nodiscard]] int64_t n_children() const;
-        [[nodiscard]] std::vector<sparrow::buffer_view<uint8_t>>& buffers();
+        void set_null_count(int64_t null_count);
+        [[nodiscard]] size_t offset() const;
+        void set_offset(size_t offset);
+        [[nodiscard]] size_t n_buffers() const;
+        void set_n_buffers(size_t n_buffers);
+        [[nodiscard]] size_t n_children() const;
+        void set_n_children(size_t n_children);
         [[nodiscard]] const std::vector<sparrow::buffer_view<uint8_t>>& buffers() const;
-        [[nodiscard]] const std::vector<arrow_proxy>& children() const;
+        [[nodiscard]] std::vector<sparrow::buffer_view<uint8_t>>& buffers();
+        void set_buffer(size_t index, const buffer_view<uint8_t>& buffer);
+        void set_buffer(size_t index, buffer<uint8_t>&& buffer);
+        [[nodiscard]] std::vector<arrow_proxy> children() const;
```

Comment:
Why returning by values instead of const reference? I don't think we want to allocate a vector each time we need to access the children.

Reply:
done